tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -898,6 +909,98 @@ def fit_extra(self, hyperparameters, time_limit=None, base_model_names=None, **k
+            test_pseudo_indices[test_pseudo_indices_true.index] = True
+
+        test_pseudo_indices = test_pseudo_indices[test_pseudo_indices]
+
+        return test_pseudo_indices
+
+    def _run_pseudo_labeling(self, hyperparameters: dict, test_data: pd.DataFrame, max_iter: int):
+        previous_score = self.info()['best_model_score_val']
+        y_pred_holdout = pd.Series()
+        X_test = test_data.copy()
+        test_data = self._learner.feature_generator.transform(test_data)
+
+        for i in range(max_iter):
+            y_pred_proba = self.predict_proba(data=X_test)
+            y_pred = self.predict(data=X_test)
+            test_pseudo_idxes_true = self._filter_pseudo(y_pred_proba_og=y_pred_proba)
+
+            if len(test_pseudo_idxes_true) < 1:
+                return self
+
+            test_pseudo_idxes = pd.Series(data=False, index=y_pred_proba.index)
+            test_pseudo_idxes[test_pseudo_idxes_true.index] = True
+
+            y_pred_holdout = y_pred_holdout.append(y_pred.loc[test_pseudo_idxes_true.index], verify_integrity=True)
+            X_pseudo = test_data.loc[y_pred_holdout.index]
+            y_pseudo = self._learner.label_cleaner.transform(y_pred_holdout)
+            self.fit_extra(hyperparameters=hyperparameters, X_pseudo=X_pseudo, y_pseudo=y_pseudo,
+                           name_suffix=PSEUDO_MODEL_SUFFIX.format(iter=(i+1)))
+            current_score = self.info()['best_model_score_val']
+
+            if previous_score >= current_score:
+                return self
+            else:
+                # Cut down X_test to not include pseudo labeled data
+                X_test = X_test.loc[test_pseudo_idxes[test_pseudo_idxes == False].index]
+                previous_score = current_score
+
+        return self
+
+    def pseudo_label_fit(self, test_data: pd.DataFrame, max_iter: int = 5, **kwargs):
+        if not self._learner.is_fit:
+            self.fit(kwargs)
+
+        X, y, _, _ = self._trainer.load_data()
+        y.name = self.label
+        train_data = pd.concat([X, y], axis=1)
+        test_data, is_labeled = self._validate_pseudo_data(train_data, test_data)
+
+        hyperparameters = None if 'hyperparameters' not in kwargs else kwargs['hyperparameters']
+        if hyperparameters is None:
+            if self._learner.is_fit:
+                hyperparameters = self.fit_hyperparameters
+
+        if isinstance(hyperparameters, str):
+            hyperparameters = get_hyperparameter_config(hyperparameters)
+
+        if is_labeled:
+            X_pseudo, y_pseudo, _, _, _, _, _, _ = self._learner.general_data_processing(test_data, None, None, 1, 0)
+            kwargs['X_pseudo'] = X_pseudo
+            kwargs['y_pseudo'] = y_pseudo
+            return self.fit_extra(hyperparameters=hyperparameters, name_suffix=PSEUDO_MODEL_SUFFIX.format(iter=1), **kwargs)
+        else:
+            return self._run_pseudo_labeling(hyperparameters=hyperparameters, test_data=test_data,
```

Comment:
why is `run_pseudolabeling` a private method? Shouldn't the user be able to use this method after they've initially called `predictor.fit()`?

Reply:
for shorter variable names, I would use "pseudolabel_" everywhere in place of "pseudo_label_"

Reply:
`pseudo_label_fit` is ideally what we want them to call because it runs the validation. `run_pseudolabeling` should be private, because there's no data validation or safety procedures in place.